banzai/qc/header_checker.py

```diff
@@ -45,144 +45,99 @@ def do_stage(self, images):
 
        """
         for image in images:
-
-            self.check_header_keyword_present(self.header_expected_format.keys(), image)
-            # Remove for now, until we can re-evaluate what exactly the
-            # expected formats should be. There could be multiple expected
-            # formats per keyword.
-            # self.check_header_format(keyword, image)
-            self.check_header_na(self.header_expected_format.keys(), image)
-            self.check_ra_range(image)
-            self.check_dec_range(image)
-            self.check_exptime_value(image)
-
+            bad_keywords = self.check_keywords_missing_or_na(image)
+            self.check_ra_range(image, bad_keywords)
+            self.check_dec_range(image, bad_keywords)
+            self.check_exptime_value(image, bad_keywords)
         return images
 
-    def check_header_keyword_present(self, keywords, image):
-        """ Logs an error if the keyword is not in the image header.
+    def check_keywords_missing_or_na(self, image):
+        """ Logs an error if the keyword right_ascension is not inside
+            the expected range (0<ra<360 degrees) in the image header.
 
         Parameters
         ----------
-        keyword : str
-                  the keyword of interest
         image : object
                 a  banzais.image.Image object.
 
+        Returns
+        -------
+        bad_keywords: list
+                a list of any keywords that are missing or NA
         """
-
         logging_tags = logs.image_config_to_tags(image, self.group_by_keywords)
+        qc_results = {}
         missing_keywords = []
-        for keyword in keywords:
+        na_keywords = []
+        for keyword in self.header_keywords_list:
             if keyword not in image.header:
                 sentence = 'The header key ' + keyword + ' is not in image header!'
+                self.logger.error(sentence, extra=logging_tags)
                 missing_keywords.append(keyword)
+            elif image.header[keyword] == 'N/A':
```

Comment:
Won't bias frames always get an RA of N/A now that we are running this on calibration frames. We should probably have a rule here about obstypes.

Reply:
Bias frames do have numerical `RA` and `DEC` values, although their `CAT-RA` and `CAT-DEC` values are usually set to `NaN`. But it is `N/A` values that specifically indicate a problem (see e.g. issue [9784](https://issues.lco.global/issues/9784))